modules/react/combobox/lib/ComboboxMenuItem.tsx

```diff
@@ -26,8 +29,22 @@ export interface ComboboxMenuItemProps {
  * added when the item has "focus" (when the cursor is on the item).
  */
 export const useComboboxMenuItem = composeHooks(
-  createElemPropsHook(useMenuModel)((model, _, elemProps: {'data-id'?: string} = {}) => {
+  createElemPropsHook(useMenuModel)((model, ref, elemProps: {'data-id'?: string} = {}) => {
     const id = elemProps['data-id'] || '';
+    const {localRef, elementRef} = useLocalRef(ref as React.Ref<HTMLElement>);
+
+    // focus on the item with the cursor
+    React.useLayoutEffect(() => {
+      if (model.state.mode === 'single') {
+        if (model.state.cursorId === id) {
+          // delay focus changes to allow PopperJS to position
+          requestAnimationFrame(() => {
+            localRef.current?.focus();
+          });
+        }
+      }
```

Comment:
```suggestion
      if (model.state.mode === 'single' && model.state.cursorId === id) {
          // delay focus changes to allow PopperJS to position
          requestAnimationFrame(() => {
            localRef.current?.focus();
          });
      }
```

No need to use nested if statements, as you have only one block